src/v0/destinations/twitter_ads/transform.js

```diff
@@ -0,0 +1,233 @@
+}
+
+// build final response
+function buildResponse(message, requestJson, metadata, endpointUrl) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpointUrl;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON.conversions = [removeUndefinedAndNullValues(requestJson)];
+  // required to be in accordance with oauth package
+  const request = {
+    url: response.endpoint,
+    method: response.method,
+    body: response.body.JSON
+  };
+  const oAuthObject = getOAuthFields(metadata);
+  let authHeader = getAuthHeaderForRequest(request, oAuthObject).Authorization;
+  if(message.properties.testModeEnable === true) {
+    authHeader = "OAuth oauth_consumer_key=\"qwe\", oauth_nonce=\"V1kMh028kZLLhfeYozuL0B45Pcx6LvuW\", oauth_signature=\"Di4cuoGv4PnCMMEeqfWTcqhvdwc%3D\", oauth_signature_method=\"HMAC-SHA1\", oauth_timestamp=\"1685603652\", oauth_token=\"yrdghfvhjvhj\", oauth_version=\"1.0\"";
+  }
+  response.headers = {
+    Authorization: authHeader,
+    'Content-Type': JSON_MIME_TYPE,
+  };
+  return response;
+}
+
+function prepareUrl(message, destination) {
+  const pixelId = message.properties.pixelId
+    ? message.properties.pixelId : destination.Config.pixelId;
+  return `${BASE_URL}/${pixelId}`;
+}
+
+function populateEventId(event, requestJson, destination) {
+
+  const eventNameToIdMappings = destination.Config.twitterAdsEventNames;
+  let eventId = "";
+
+  if (eventNameToIdMappings) {
+    eventNameToIdMappings.forEach(obj => {
+      if (obj.rudderEventName && obj.rudderEventName.trim() && obj.rudderEventName.trim().toLowerCase() == event.toString().toLowerCase()) {
+        eventId = obj.twitterEventId;
+      }
+    });
+  }
+
+  if(!eventId) {
+    throw new ConfigurationError(`[TWITTER ADS]: Event - '${event}' do not have a corresponding eventId in configuration. Aborting`);
+  }
+
+  return eventId;
+}
+
+// process track call
+function processTrack(message, metadata, destination) {
+
+  const requestJson = constructPayload(message, mappingConfig[ConfigCategories.TRACK.name]);
+
+  requestJson.event_id = requestJson.event_id || populateEventId(message.event, requestJson, destination);
+
+  requestJson.conversion_time = isDefinedAndNotNull(requestJson.conversion_time)
+    ? requestJson.conversion_time : message.timestamp;
+
+  const identifiers = [];
+
+  if (message.properties.email) {
+    const obj = {};
+    let email = message.properties.email.trim();
+    if (email) {
+      email = email.toLowerCase();
+      obj.hashed_email = sha256(email);
+      identifiers.push(obj)
+    }
+  }
+
+  if (message.properties.phone) {
+    const obj = {};
+    let phone = message.properties.phone.trim();
+    if (phone) {
+      obj.hashed_phone_number = sha256(phone);
+      identifiers.push(obj)
+    }
+  }
+
+  if (message.properties.twclid) {
+    const obj = {};
+    obj.twclid = sha256(message.properties.twclid);
+    identifiers.push(obj);
+  }
```

Comment:
some operations are repetitive here, it can be improved

const hashValue = (value) => {
  // Replace this with your preferred hashing algorithm
  return sha256(value);
};

const trimAndLowercase = (value) => {
  return value.trim().toLowerCase();
};

const identifiers = [];

if (message.properties.email) {
  let email = trimAndLowercase(message.properties.email);
  if (email) {
    identifiers.push({ hashed_email: hashValue(email) });
  }
}

if (message.properties.phone) {
  let phone = trimAndLowercase(message.properties.phone);
  if (phone) {
    identifiers.push({ hashed_phone_number: hashValue(phone) });
  }
}

if (message.properties.twclid) {
  identifiers.push({ twclid: hashValue(message.properties.twclid) });
}

Reply:
hashValue is unneccesary as we are only calling sha method os used it directly. Other changes were done as part of eslint fixes